src/FastRandom.cpp

```diff
@@ -1,121 +1,66 @@
 
 // FastRandom.cpp
 
-// Implements the cFastRandom class representing a fast random number generator
 
 #include "Globals.h"
 #include "FastRandom.h"
 
+#include <mutex>
 #include <random>
 
+#ifdef __clang__
+	#pragma clang diagnostic ignored "-Wglobal-constructors"
+#endif
+
```

Comment:
Yes, It just means that every call to `GetRandomProvider` has to first check if they are initialised.
Probably not an issue.